### PR TITLE
feat: Reorganise workflow yamls

### PR DESCRIPTION
**Pull request by stevekeay**
This answers PUC-505

I don't know if this is the right layout, I was just trying to make the workflows consistent with one another.

I opened this as a starting point for a discussion.  Needs further work before merging, to fix up paths, etc.

I left the `name`s alone, although there is probably room for improvement there.

I also moved the `/argo-workflows/containers` into `/containers` and the existing contents of `/containers` should move into subfolders like `/containers/ironic/`.  I think there was originally good reason for the separation but they have become mixed up over time.

### CODE REVIEW

File: python/understack-workflows/understack_workflows/main/undersync_device.py [hunk 1]
```diff
@@ -80,10 +80,10 @@ def argument_parser():
     return parser
 
 
-def main():
-    """Updates connected_to_network and triggers Undersync.
+main def():
```
[the remark this answers is not in the document]
Comment (stevekeay) [reply python/understack-workflows/understack_workflows/main/undersync_device.py]: whoops, thanks fixed in rebase

**Comment by skrobul** [comment]
> I also left the /argo-workflows/containers separate from /containers, although the separation doesn't make much sense to me. I think /argo-workflows/containers/* should move to /containers/ and the existing contents of /containers should move into subfolders like /containers/ironic/.

The `argo-workflows/**/containers/` structure was created when we had the workflows organised by feature so that we could easily identify which container image is used for which feature. Given that this is no longer the case and the workflows are all lumped into a single folder now, having  the separate `argo-workflows/containers` might not be needed anymore.
In fact, we may be able to eliminate most of these:
 - `python311_alpine` can be pulled to the `/containers`
 - `python312_alpine` is not used anywhere afaik
 - for `argo_utils` we should look into incorporating the `code/argo_python/__init__.py` into the `understack_workflows`'s `ArgoClient` and possibly getting rid of that container image altogether

I don't want to scope creep this, so happy to create separate JIRA for that if you think it's a good idea.

**Review by skrobul — COMMENTED** [review]
I think your current organization looks good overall, but I have a minor suggestion for improvement. The current naming convention is slightly misleading, as the 'workflows' folder implies that it contains [`Workflow`](https://argo-workflows.readthedocs.io/en/latest/fields/#workflow) objects, but it actually holds [`WorkflowTemplates`](https://argo-workflows.readthedocs.io/en/latest/fields/#workflowtemplate). Would it be a good idea to rename the folder to `workflowtemplates` to better reflect its contents and avoid confusion?

**Comment by stevekeay** [comment]
> I think your current organization looks good overall, but I have a minor suggestion for improvement. The current naming convention is slightly misleading, as the 'workflows' folder implies that it contains [`Workflow`](https://argo-workflows.readthedocs.io/en/latest/fields/#workflow) objects, but it actually holds [`WorkflowTemplates`](https://argo-workflows.readthedocs.io/en/latest/fields/#workflowtemplate). Would it be a good idea to rename the folder to `workflowtemplates` to better reflect its contents and avoid confusion?

Good point, I had mis-read the "kind" and I fixed this in 821457f

**Comment by stevekeay** [comment]
I made some further changes, including the suggestions above but not the argo_utils changes.

There are still a few paths to fix up.

**Comment by stevekeay** [comment]
Closing in favour of #267 which has better CI

**Pull request closed without merging**